### PR TITLE
New datagrid view for product variant management

### PR DESCRIPTION
**Pull request by dominik-zeglen**
I want to merge this change because it adds datagrid as variant management component.

**PR intended to be tested with API branch:** main

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
MARKETPLACE_URL=https://marketplace-gray.vercel.app/

### Do you want to run more stable tests?
Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation

CONTAINERS=1

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-6661 Datagrid](https://app.clickup.com/t/2549495/SALEOR-6661)

**Review by orzechdev — APPROVED** [review]

**Review by Droniu — APPROVED** [review]
Snapshots & extract-messages before merging

**Review by eCzerniak — DISMISSED** [review]

File: src/components/Datagrid/usePreventHistoryBack.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import throttle from "lodash/throttle";
+import React from "react";
+
```
Comment (Droniu) [inline src/components/Datagrid/usePreventHistoryBack.ts:3]: This has a potential to be reusable, we could add jsdoc-style comment how to use it and put it in `src/hooks`

**Review by Droniu — APPROVED** [review]
LGTM

File: src/components/Datagrid/Datagrid.tsx [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+                            }}
+                            menuItems={menuItems(index)}
+                          />
+                        </div>
+                      ))}
+                  </div>
+                }
+                rowMarkerWidth={48}
+              />
+              {/* FIXME: https://github.com/glideapps/glide-data-grid/issues/505 */}
```
Comment (orzechdev) [inline src/components/Datagrid/Datagrid.tsx:316]: In such a cases when we left `FIXME` or `TODO`, shouldn't we also make an issue in our dashboard repo and link it in the comment to keep track of it?

File: src/components/Grid/Grid.tsx [hunk 3]
```diff
@@ -33,11 +33,8 @@ const useStyles = makeStyles(
     uniform: {
       gridTemplateColumns: "1fr 1fr",
     },
-    richText: {
-      "&& > div": {
-        overflow: "visible",
-      },
-    },
+    // TODO: Remove me
+    richText: {},
```
Comment (orzechdev) [inline src/components/Grid/Grid.tsx:37]: What is this?

**Review by orzechdev — DISMISSED** [review]

File: src/components/Datagrid/Datagrid.tsx (hunk 2 above)
Comment (andrzejewsky) [reply src/components/Datagrid/Datagrid.tsx:316]: will do, I have many things that I will create tasks for

**Review by Droniu — DISMISSED** [review]

**Review by orzechdev — DISMISSED** [review]

**Review by orzechdev — APPROVED** [review]

**Review by Droniu — APPROVED** [review]

**Review by eCzerniak — APPROVED** [review]

**Pull request merged**